sdks/python/apache_beam/io/fileio.py

```diff
@@ -249,14 +230,11 @@ def __exit__(self, exception_type, exception_value, traceback):
 
   def __iter__(self):
     if self.current_offset > 0 and ChannelFactory.is_compressed(self.file):
-      # When compression is enabled both initial and dynamic splitting should be
-      # prevented. Here we prevent initial splitting by ignoring all splits
-      # other than the split that starts at byte 0.
-      #
-      # TODO: Turns this warning into an exception soon.
-      logging.warning('Ignoring split starting at (%s) for compressed source.',
-                      self.current_offset)
-      return
+      # When compression is enabled both initial and dynamic splitting should
+      # not be allowed.
+      raise ValueError(
```

Comment:
Based on old and recent discussions with Chamikara, this would only happen if:
a) The Service introduces initial splits for compressed sources (which it shouldn't do, or else it's a bug).
or
b) The worker dynamically splits a compressed source (which it shouldn't do, or else it's a bug).

The various unit and integration tests (both internal and external) passed, which suggests that both the service and the worker are behaving correctly.

I'm open to suggestions of extra testing as well.